goldmane/fv/daemon_test.go

```diff
@@ -290,3 +290,72 @@ func TestHints(t *testing.T) {
 		return nil
 	}, 5*time.Second, 1*time.Second).Should(Succeed())
 }
+
+func TestStatistics(t *testing.T) {
+	cfg := daemon.Config{
+		LogLevel:                 "debug",
+		Port:                     8988,
+		AggregationWindow:        time.Second * 1,
+		EmitAfterSeconds:         2,
+		EmitterAggregationWindow: time.Second * 2,
+	}
+	defer daemonSetup(t, cfg)()
+
+	// Generate credentials for the Goldmane client.
+	creds, err := client.ClientCredentials(clientCert, clientKey, clientCA)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to create goldmane TLS credentials.")
+	}
+
+	// Generate credentials for the Goldmane client.
+	statsClient, err := client.NewStatisticsAPIClient(goldmaneURL, grpc.WithTransportCredentials(creds))
+	require.NoError(t, err)
+
+	// Send a request for statistics with no data.
+	Eventually(func() error {
+		res, err := statsClient.List(ctx, &proto.StatisticsRequest{})
+		if err != nil {
+			return err
+		}
+		if res != nil {
```

Comment:
Is it enough to check only for `nill`? should we also check for empty object? If I'm not mistaken GRPC returns  zero-value struct (not nill) if there’s "no data".

Reply:
Yep, good catch. This should be a `len(res)` check.

Reply:
FWIW, I believe this code actually _does_ work based on how the List() function is implemented: https://github.com/projectcalico/calico/blob/7fe4acde84f2450ef720165f1816f2ede87a5325/goldmane/pkg/client/statistics_service.go#L54-L65

Note that if there are no results from the stream we'll return an uninitialized `[]*proto.StatisticsResult`, which will evaluate to `nil`. 

Still better to improve the check though.

Reply:
I see :ok_hand: